PowerEditor/src/ScintillaComponent/AutoCompletion.cpp

```diff
@@ -58,6 +118,8 @@ bool AutoCompletion::showApiComplete()
 		return false;
 
 	_pEditView->execute(SCI_AUTOCSETSEPARATOR, WPARAM(' '));
+	_pEditView->execute(SCI_AUTOCSETTYPESEPARATOR, WPARAM('?'));
+	_pEditView->execute(SCI_REGISTERIMAGE, FUNC_IMG_ID, LPARAM(xpmfn));

```

Comment:
Here `SCI_REGISTERIMAGE` are called each time  `showApiComplete` is called.
Whereas we need only call it once for all (tested).
I guess it's the same case for both `SCI_AUTOCSETSEPARATOR` & `SCI_AUTOCSETTYPESEPARATOR`, but I didn't tested yet.

Reply:
Yes, since `SCI_AUTOCSETSEPARATOR` was already there, I assumed adding the others in the same would work.  This could be considered defensive in case some plugin changes the `SCI_AUTOCSETSEPARATOR` or registers an image with the same ID.  Off chance on the latter, but I use autocomplete in 2 plugins (QuickText, TagLEET) as well as some PythonScripts.  I always keep the `SCI_AUTOCSETSEPARATOR` = ` " "` and `SCI_AUTOCSETTYPESEPARATOR` = `?`; others - who knows?

Cheers.